course6/toyapp-adam/math/vector.h

```diff
@@ -116,6 +122,7 @@ class Vector
 		return Vector{*this} -= rhs;
 	}
 
+	template<typename FloatType = float, typename = IsFloatingPoint<FloatType>>
```

Comment:
What's the point of making this a template?

Reply:
It uses `Length()` internally, which is a template.